src/providers/wms/qgswmsdataitems.cpp

```diff
@@ -365,6 +365,18 @@ QString QgsWMSLayerItem::createUri()
   QString style = !mLayerProperty.style.isEmpty() ? mLayerProperty.style.at( 0 ).name : QString();
   mDataSourceUri.setParam( QStringLiteral( "styles" ), style );
 
+  // Check for layer dimensions
+  for ( const QgsWmsDimensionProperty &dimension : qgis::as_const( mLayerProperty.dimensions ) )
+  {
+    // add temporal dimensions only
+    if ( dimension.name == "time" || dimension.name == "reference_time" )
```

Comment:
```suggestion
    if ( dimension.name == QLatin1String( "time" ) || dimension.name ==  QLatin1String( "reference_time" ) )
```

(and also the checks below too)